File: backend/microservice/src/level/objects/Rating.scala
package microservice.level.objects

import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}

/** 玩家对关卡的评分记录。
  *
  * 字段说明：
  *   - id / levelId / playerId：评分标识与关联
  *   - score：1–5 分
  *   - createdAt / updatedAt：首次评分与最后更新时间
  */
final case class Rating(
  id: String,
  levelId: String,
  playerId: String,
  score: Int,
  createdAt: String,
  updatedAt: String
)

object Rating {
  implicit val encoder: Encoder[Rating] = deriveEncoder
  implicit val decoder: Decoder[Rating] = deriveDecoder
}

File: frontend/src/objects/level/rating.ts
import { z } from "zod";

export const RatingSchema = z.object({
  id: z.string().min(1),
  levelId: z.string().min(1),
  playerId: z.string().min(1),
  score: z.number().int().min(1).max(5),
  createdAt: z.string(),
  updatedAt: z.string(),
});

export type Rating = z.infer<typeof RatingSchema>;
